src/CoreShop/Component/Core/Order/Processor/CartRuleAutoRemoverProcessor.php

```diff
@@ -78,7 +80,10 @@ public function process(CartInterface $cart)
 
             $voucherCode = $this->voucherCodeRepository->findByCode($item->getVoucherCode());
 
-            if (!$this->cartPriceRuleValidator->isValidCartRule($cart, $item->getCartPriceRule(), $voucherCode)) {
+            if (!$item->getCartPriceRule()->getIsVoucherRule() && null !== $item->getVoucherCode()) {
```

Comment:
this removes rules that are not voucher rules but have a voucher code in the cart